Support Java 9

I know Webstart will disappear, but I want to see if I can compile and run with it under Java 9.

Is there parameters I should add to the java compiler so it can work ?

[INFO] --- webstart-maven-plugin:1.0-beta-7:jnlp (default) @ client ---

[WARNING] Error injecting: org.codehaus.mojo.webstart.sign.DefaultSignTool
java.lang.NoClassDefFoundError: Lorg/codehaus/mojo/keytool/KeyTool;
	at java.base/java.lang.Class.getDeclaredFields0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredFields(Class.java:3024)
	at java.base/java.lang.Class.getDeclaredFields(Class.java:2205)
	at com.google.inject.spi.InjectionPoint.getInjectionPoints(InjectionPoint.java:675) 
...

Caused by: java.lang.ClassNotFoundException: org.codehaus.mojo.keytool.KeyTool
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
...

Caused by: java.lang.ClassNotFoundException: org.codehaus.mojo.keytool.KeyTool
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
...

--- a/.github/workflows/jdkbuilds.yml
+++ b/.github/workflows/jdkbuilds.yml
@@ -13,7 +13,7 @@ jobs:
   openjdk:
     strategy:
       matrix:
-        jdk: [8-slim]
+        jdk: [8-slim, 11-slim]
     name: "OpenJDK ${{ matrix.jdk }}"
     runs-on: ubuntu-latest
     container: "maven:3.6.2-jdk-${{ matrix.jdk }}"

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,6 +6,7 @@ install: true
 jdk: 
   - openjdk7
   - openjdk8
+  - openjdk11
 
 script:
   - mvn clean verify -B -e -V

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,9 @@
 dist: trusty
 language: java
-sudo: false
 install: true
+
+os:
+  - linux
 
 jdk: 
   - openjdk7
@@ -11,9 +13,9 @@ jdk:
 script:
   - mvn clean verify -B -e -V
 
-cache:
-  directories:
-    - '$HOME/.m2/repository'
+#cache:
+#  directories:
+#    - '$HOME/.m2/repository'
 
 branches:
     except:

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,4 +1,4 @@
-dist: trusty
+dist: xenial
 language: java
 install: true
 
@@ -10,12 +10,17 @@ jdk:
   - openjdk8
   - openjdk11
 
+env:
+  global:
+    - MVN_VERSION="3.6.3"
+
 script:
+  - mvn -version
   - mvn clean verify -B -e -V
 
-#cache:
-#  directories:
-#    - '$HOME/.m2/repository'
+cache:
+  directories:
+    - '$HOME/.m2/repository'
 
 branches:
     except:

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,7 +6,6 @@ os:
   - linux
 
 jdk: 
-  - openjdk7
   - openjdk8
   - openjdk11
 

--- a/pom.xml
+++ b/pom.xml
@@ -103,9 +103,10 @@
   </distributionManagement>
 
   <properties>
-    <mavenVersion>2.0.11</mavenVersion>
+    <mavenVersion>3.5.4</mavenVersion>
     <mavenPluginPluginVersion>3.2</mavenPluginPluginVersion>
     <keytoolVersion>1.5</keytoolVersion>
+	<plexusComponentVersion>1.7.1</plexusComponentVersion>
 
     <!-- release plugin configuration -->
 
@@ -128,11 +129,6 @@
     <dependencies>
       <dependency>
         <groupId>org.apache.maven</groupId>
-        <artifactId>maven-model</artifactId>
-        <version>${mavenVersion}</version>
-      </dependency>
-      <dependency>
-        <groupId>org.apache.maven</groupId>
         <artifactId>maven-artifact</artifactId>
         <version>${mavenVersion}</version>
       </dependency>
@@ -152,11 +148,11 @@
       <dependency>
         <groupId>org.codehaus.plexus</groupId>
         <artifactId>plexus-component-annotations</artifactId>
-        <version>1.5.5</version>
+        <version>${plexusComponentVersion}</version>
       </dependency>
       <dependency>
         <groupId>org.apache.maven</groupId>
-        <artifactId>maven-project</artifactId>
+        <artifactId>maven-compat</artifactId>
         <version>${mavenVersion}</version>
       </dependency>
       <dependency>
@@ -185,7 +181,7 @@
       <dependency>
         <groupId>org.apache.maven.reporting</groupId>
         <artifactId>maven-reporting-api</artifactId>
-        <version>${mavenVersion}</version>
+        <version>3.0</version>
         <exclusions>
           <exclusion>
             <groupId>plexus</groupId>
@@ -196,7 +192,7 @@
       <dependency>
         <groupId>org.apache.maven.reporting</groupId>
         <artifactId>maven-reporting-impl</artifactId>
-        <version>2.0.4</version>
+        <version>3.0.0</version>
         <exclusions>
           <exclusion>
             <groupId>plexus</groupId>
@@ -208,7 +204,7 @@
       <dependency>
         <groupId>org.apache.maven.shared</groupId>
         <artifactId>maven-shared-utils</artifactId>
-        <version>0.5</version>
+        <version>3.2.1</version>
       </dependency>
 
       <dependency>
@@ -221,7 +217,7 @@
       <dependency>
         <groupId>org.apache.maven.shared</groupId>
         <artifactId>maven-jarsigner</artifactId>
-        <version>1.3.1</version>
+        <version>3.0.0</version>
         <exclusions>
           <exclusion>
             <groupId>org.codehaus.plexus</groupId>
@@ -249,21 +245,7 @@
         <groupId>org.codehaus.plexus</groupId>
         <artifactId>plexus-utils</artifactId>
         <version>3.0.10</version>
-        <!--<version>1.4.9</version>-->
       </dependency>
-
-      <dependency>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>keytool-api-1.5</artifactId>
-        <version>${keytoolVersion}</version>
-      </dependency>
-
-      <dependency>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>keytool-api-1.6</artifactId>
-        <version>${keytoolVersion}</version>
-      </dependency>
-
       <dependency>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>keytool-api-1.7</artifactId>
@@ -273,7 +255,7 @@
       <dependency>
         <groupId>javax.servlet</groupId>
         <artifactId>servlet-api</artifactId>
-        <version>2.3</version>
+        <version>2.4</version>
       </dependency>
 
       <dependency>
@@ -329,7 +311,7 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
-				</plugin>
+		</plugin>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-plugin-plugin</artifactId>
@@ -374,6 +356,24 @@
           <configuration>
             <content>${project.build.directory}/staging/${project.artifactId}</content>
           </configuration>
+        </plugin>
+		<plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-invoker-plugin</artifactId>
+          <version>3.2.1</version>
+          <configuration>
+            <scriptVariables>
+              <projectVersion>${project.version}</projectVersion>
+            </scriptVariables>
+            <environmentVariables>
+              <MAVEN_OPTS>-Duser.country=US -Duser.language=en</MAVEN_OPTS>
+            </environmentVariables>
+          </configuration>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>2.22.2</version>
         </plugin>
       </plugins>
     </pluginManagement>

--- a/pom.xml
+++ b/pom.xml
@@ -261,8 +261,7 @@
       <dependency>
         <groupId>junit</groupId>
         <artifactId>junit</artifactId>
-        <version>4.9</version>
-        <scope>test</scope>
+        <version>4.13.1</version>
       </dependency>
       <dependency>
         <groupId>xmlunit</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -104,9 +104,11 @@
 
   <properties>
     <mavenVersion>3.5.4</mavenVersion>
-    <mavenPluginPluginVersion>3.2</mavenPluginPluginVersion>
+    <mavenPluginPluginVersion>3.6.0</mavenPluginPluginVersion>
     <keytoolVersion>1.5</keytoolVersion>
 	<plexusComponentVersion>1.7.1</plexusComponentVersion>
+
+	<velocity-version>2.2</velocity-version>
 
     <!-- release plugin configuration -->
 
@@ -229,7 +231,7 @@
       <dependency>
         <groupId>org.codehaus.plexus</groupId>
         <artifactId>plexus-archiver</artifactId>
-        <version>2.4.1</version>
+        <version>4.2.3</version>
         <exclusions>
           <exclusion>
             <groupId>org.codehaus.plexus</groupId>
@@ -244,7 +246,7 @@
       <dependency>
         <groupId>org.codehaus.plexus</groupId>
         <artifactId>plexus-utils</artifactId>
-        <version>3.0.10</version>
+        <version>3.3.0</version>
       </dependency>
       <dependency>
         <groupId>org.codehaus.mojo</groupId>
@@ -266,32 +268,26 @@
       <dependency>
         <groupId>xmlunit</groupId>
         <artifactId>xmlunit</artifactId>
-        <version>1.2</version>
+        <version>1.6</version>
       </dependency>
 
       <dependency>
-        <groupId>commons-lang</groupId>
-        <artifactId>commons-lang</artifactId>
-        <version>2.6</version>
-      </dependency>
-     
-      <dependency>
         <groupId>org.apache.velocity</groupId>
-        <artifactId>velocity</artifactId>
-        <version>1.7</version>
+        <artifactId>velocity-engine-core</artifactId>
+        <version>${velocity-version}</version>
       </dependency>
       
       <dependency>
-        <groupId>org.apache.maven.shared</groupId>
+        <groupId>org.apache.maven.plugin-testing</groupId>
         <artifactId>maven-plugin-testing-harness</artifactId>
-        <version>1.1</version>
+        <version>3.3.0</version>
         <scope>test</scope>
       </dependency>
 
       <dependency>
         <groupId>org.codehaus.plexus</groupId>
         <artifactId>plexus-container-default</artifactId>
-        <version>1.0-alpha-9</version>
+        <version>2.1.0</version>
       </dependency>
 
       <!-- for JarUnsign... -->

--- a/pom.xml
+++ b/pom.xml
@@ -109,6 +109,10 @@
 	<plexusComponentVersion>1.7.1</plexusComponentVersion>
 
 	<velocity-version>2.2</velocity-version>
+	
+	<!-- maven.compiler.compilerVersion>1.8</maven.compiler.compilerVersion -->
+	<maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
 
     <!-- release plugin configuration -->
 
@@ -306,6 +310,11 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
+			<configuration>
+				<!-- compilerVersion>${maven.compiler.compilerVersion}</compilerVersion -->
+				<source>${maven.compiler.source}</source>
+				<target>${maven.compiler.target}</target>
+			</configuration>
 		</plugin>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
@@ -393,4 +402,16 @@
       </plugin>
     </plugins>
   </build>
+	<profiles>
+		<profile>
+			<id>jdk11</id>
+			<activation>
+				<jdk>11</jdk>
+			</activation>
+			<properties>
+				<maven.compiler.release>8</maven.compiler.release>
+			</properties>
+		</profile>	
+	</profiles>
+  
 </project>

--- a/webstart-maven-plugin/pom.xml
+++ b/webstart-maven-plugin/pom.xml
@@ -38,11 +38,11 @@
   </prerequisites>
 
   <properties>
-    <javadocPluginVersion>2.7</javadocPluginVersion>
+    <javadocPluginVersion>3.2.0</javadocPluginVersion>
 
     <!-- versions of javadoc tagglets -->
-    <mavenPluginToolsJavadocVersion>2.5.1</mavenPluginToolsJavadocVersion>
-    <plexusComponentJavadocVersion>1.3.0</plexusComponentJavadocVersion>
+    <mavenPluginToolsJavadocVersion>3.6.0</mavenPluginToolsJavadocVersion>
+    <plexusComponentJavadocVersion>2.1.0</plexusComponentJavadocVersion>
   </properties>
 
   <dependencies>
@@ -133,13 +133,9 @@
 
     <dependency>
       <groupId>org.apache.velocity</groupId>
-      <artifactId>velocity</artifactId>
+      <artifactId>velocity-engine-core</artifactId>
     </dependency>
 
-    <dependency>
-      <groupId>commons-lang</groupId>
-      <artifactId>commons-lang</artifactId>
-    </dependency>
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
@@ -159,7 +155,7 @@
     </dependency>
 
     <dependency>
-      <groupId>org.apache.maven.shared</groupId>
+      <groupId>org.apache.maven.plugin-testing</groupId>
       <artifactId>maven-plugin-testing-harness</artifactId>
       <scope>test</scope>
     </dependency>

--- a/webstart-maven-plugin/pom.xml
+++ b/webstart-maven-plugin/pom.xml
@@ -47,10 +47,11 @@
 
   <dependencies>
 
-    <dependency>
-      <groupId>org.apache.maven</groupId>
-      <artifactId>maven-model</artifactId>
-    </dependency>
+	<dependency>
+	  <groupId>org.codehaus.mojo</groupId>
+	  <artifactId>keytool-api-1.7</artifactId>
+	</dependency>
+
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-artifact</artifactId>
@@ -59,10 +60,11 @@
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
     </dependency>
-    <dependency>
-      <groupId>org.apache.maven</groupId>
-      <artifactId>maven-project</artifactId>
-    </dependency>
+      <dependency>
+        <groupId>org.apache.maven</groupId>
+        <artifactId>maven-compat</artifactId>
+      </dependency>
+
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
@@ -243,54 +245,6 @@
 
   <profiles>
     <profile>
-      <id>jdk18</id>
-      <activation>
-        <jdk>1.8</jdk>
-      </activation>
-      <dependencies>
-        <dependency>
-          <groupId>org.codehaus.mojo</groupId>
-          <artifactId>keytool-api-1.7</artifactId>
-        </dependency>
-      </dependencies>
-    </profile>
-    <profile>
-      <id>jdk17</id>
-      <activation>
-        <jdk>1.7</jdk>
-      </activation>
-      <dependencies>
-        <dependency>
-          <groupId>org.codehaus.mojo</groupId>
-          <artifactId>keytool-api-1.7</artifactId>
-        </dependency>
-      </dependencies>
-    </profile>
-    <profile>
-      <id>jdk16</id>
-      <activation>
-        <jdk>1.6</jdk>
-      </activation>
-      <dependencies>
-        <dependency>
-          <groupId>org.codehaus.mojo</groupId>
-          <artifactId>keytool-api-1.6</artifactId>
-        </dependency>
-      </dependencies>
-    </profile>
-    <profile>
-      <id>jdk15</id>
-      <activation>
-        <jdk>1.5</jdk>
-      </activation>
-      <dependencies>
-        <dependency>
-          <groupId>org.codehaus.mojo</groupId>
-          <artifactId>keytool-api-1.5</artifactId>
-        </dependency>
-      </dependencies>
-    </profile>
-    <profile>
       <id>run-its</id>
       <activation>
         <property>
@@ -345,7 +299,7 @@
 
           <plugin>
             <artifactId>maven-invoker-plugin</artifactId>
-						<version>3.2.1</version>
+			<version>3.2.1</version>
           </plugin>
           <plugin>
             <groupId>org.codehaus.mojo</groupId>

--- a/webstart-maven-plugin/src/it/MWEBSTART-11/pom.xml
+++ b/webstart-maven-plugin/src/it/MWEBSTART-11/pom.xml
@@ -98,6 +98,15 @@
         </configuration>
       </plugin>
     </plugins>
+	<pluginManagement>
+	  <plugins>
+		<plugin>
+			<groupId>org.apache.maven.plugins</groupId>
+			<artifactId>maven-compiler-plugin</artifactId>
+			<version>3.8.1</version>
+		</plugin>
+	  </plugins>
+	</pluginManagement>
   </build>
 
   <dependencies>

--- a/webstart-maven-plugin/src/it/MWEBSTART-138/jnlp-dep/pom.xml
+++ b/webstart-maven-plugin/src/it/MWEBSTART-138/jnlp-dep/pom.xml
@@ -16,8 +16,8 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
         <configuration>
-          <source>1.5</source>
-          <target>1.5</target>
+          <source>1.8</source>
+          <target>1.8</target>
           <encoding>UTF-8</encoding>
         </configuration>
       </plugin>

--- a/webstart-maven-plugin/src/it/MWEBSTART-138/pom.xml
+++ b/webstart-maven-plugin/src/it/MWEBSTART-138/pom.xml
@@ -52,6 +52,11 @@
             </sign>
           </configuration>
         </plugin>
+		<plugin>
+			<groupId>org.apache.maven.plugins</groupId>
+			<artifactId>maven-compiler-plugin</artifactId>
+			<version>3.8.1</version>
+		</plugin>
       </plugins>
     </pluginManagement>
   </build>

--- a/webstart-maven-plugin/src/it/MWEBSTART-138/real-dep/pom.xml
+++ b/webstart-maven-plugin/src/it/MWEBSTART-138/real-dep/pom.xml
@@ -16,8 +16,8 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
         <configuration>
-          <source>1.5</source>
-          <target>1.5</target>
+          <source>1.8</source>
+          <target>1.8</target>
           <encoding>UTF-8</encoding>
         </configuration>
       </plugin>

--- a/webstart-maven-plugin/src/it/MWEBSTART-219/pom.xml
+++ b/webstart-maven-plugin/src/it/MWEBSTART-219/pom.xml
@@ -17,4 +17,16 @@
     <module>webstart</module>
   </modules>
 
+  <build>
+    <pluginManagement>
+      <plugins>
+		<plugin>
+			<groupId>org.apache.maven.plugins</groupId>
+			<artifactId>maven-compiler-plugin</artifactId>
+			<version>3.8.1</version>
+		</plugin>
+      </plugins>
+    </pluginManagement>
+  </build>
+
 </project>

--- a/webstart-maven-plugin/src/it/it001/pom.xml
+++ b/webstart-maven-plugin/src/it/it001/pom.xml
@@ -17,6 +17,11 @@
 
   <build>
     <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.8.1</version>
+        </plugin>
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>webstart-maven-plugin</artifactId>
@@ -94,13 +99,6 @@
       <version>2.1</version>
     </dependency>
     <dependency>
-      <groupId>com.sun</groupId>
-      <artifactId>tools</artifactId>
-      <version>1.4.2</version>
-      <scope>system</scope>
-      <systemPath>${toolsjar}</systemPath>
-    </dependency>
-    <dependency>
       <groupId>javax.servlet</groupId>
       <artifactId>servlet-api</artifactId>
       <version>2.3</version>
@@ -108,28 +106,4 @@
     </dependency>
   </dependencies>
 
-  <profiles>
-    <profile>
-      <id>default-profile</id>
-      <activation>
-        <file>
-          <exists>${java.home}/../lib/tools.jar</exists>
-        </file>
-      </activation>
-      <properties>
-        <toolsjar>${java.home}/../lib/tools.jar</toolsjar>
-      </properties>
-    </profile>
-    <profile>
-      <id>mac-profile</id>
-      <activation>
-        <file>
-          <exists>${java.home}/../Classes/classes.jar</exists>
-        </file>
-      </activation>
-      <properties>
-        <toolsjar>${java.home}/../Classes/classes.jar</toolsjar>
-      </properties>
-    </profile>
-  </profiles>
 </project>

--- a/webstart-maven-plugin/src/it/it002/core/pom.xml
+++ b/webstart-maven-plugin/src/it/it002/core/pom.xml
@@ -20,13 +20,6 @@
       <artifactId>junit</artifactId>
     </dependency>
 
-    <!-- these dependencies are not really needed for this module to build, 
-         just added to check that dependencies are properly filtered out -->
-    <dependency>
-      <groupId>com.sun</groupId>
-      <artifactId>tools</artifactId>
-    </dependency>
-
     <dependency>
       <groupId>javax.servlet</groupId>
       <artifactId>servlet-api</artifactId>

--- a/webstart-maven-plugin/src/it/it002/hello-world/pom.xml
+++ b/webstart-maven-plugin/src/it/it002/hello-world/pom.xml
@@ -25,13 +25,6 @@
       <artifactId>junit</artifactId>
     </dependency>
 
-    <!-- these dependencies are not really needed for this module to build, 
-         just added to check that dependencies are properly filtered out -->
-    <dependency>
-      <groupId>com.sun</groupId>
-      <artifactId>tools</artifactId>
-    </dependency>
-
     <dependency>
       <groupId>javax.servlet</groupId>
       <artifactId>servlet-api</artifactId>

--- a/webstart-maven-plugin/src/it/it002/pom.xml
+++ b/webstart-maven-plugin/src/it/it002/pom.xml
@@ -28,7 +28,7 @@
       <dependency>
         <groupId>junit</groupId>
         <artifactId>junit</artifactId>
-        <version>3.8.1</version>
+        <version>4.13.1</version>
         <scope>test</scope>
       </dependency>
 

--- a/webstart-maven-plugin/src/it/it002/pom.xml
+++ b/webstart-maven-plugin/src/it/it002/pom.xml
@@ -32,16 +32,6 @@
         <scope>test</scope>
       </dependency>
 
-      <!-- these dependencies are not really needed for this module to build,
-just added to check that dependencies are properly filtered out -->
-      <dependency>
-        <groupId>com.sun</groupId>
-        <artifactId>tools</artifactId>
-        <version>1.4.2</version>
-        <scope>system</scope>
-        <systemPath>${toolsjar}</systemPath>
-      </dependency>
-
       <dependency>
         <groupId>javax.servlet</groupId>
         <artifactId>servlet-api</artifactId>
@@ -62,6 +52,11 @@ just added to check that dependencies are properly filtered out -->
         </plugin>
 
         <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>3.8.1</version>
+        </plugin>
+        <plugin>
             <groupId>org.mortbay.jetty</groupId>
             <artifactId>maven-jetty-plugin</artifactId>
             <version>6.1.6</version>
@@ -69,29 +64,4 @@ just added to check that dependencies are properly filtered out -->
       </plugins>
     </pluginManagement>
   </build>
-
-  <profiles>
-    <profile>
-      <id>default-profile</id>
-      <activation>
-        <file>
-          <exists>${java.home}/../lib/tools.jar</exists>
-        </file>
-      </activation>
-      <properties>
-        <toolsjar>${java.home}/../lib/tools.jar</toolsjar>
-      </properties>
-    </profile>
-    <profile>
-      <id>mac-profile</id>
-      <activation>
-        <file>
-          <exists>${java.home}/../Classes/classes.jar</exists>
-        </file>
-      </activation>
-      <properties>
-        <toolsjar>${java.home}/../Classes/classes.jar</toolsjar>
-      </properties>
-    </profile>
-  </profiles>
 </project>

--- a/webstart-maven-plugin/src/it/it003/pom.xml
+++ b/webstart-maven-plugin/src/it/it003/pom.xml
@@ -28,7 +28,7 @@
       <dependency>
         <groupId>junit</groupId>
         <artifactId>junit</artifactId>
-        <version>3.8.1</version>
+        <version>4.13.1</version>
         <scope>test</scope>
       </dependency>
 

--- a/webstart-maven-plugin/src/it/it003/pom.xml
+++ b/webstart-maven-plugin/src/it/it003/pom.xml
@@ -49,6 +49,11 @@
           <artifactId>webstart-maven-plugin</artifactId>
           <version>@project.version@</version>
         </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.8.1</version>
+        </plugin>
       </plugins>
     </pluginManagement>
   </build>

--- a/webstart-maven-plugin/src/it/it005/pom.xml
+++ b/webstart-maven-plugin/src/it/it005/pom.xml
@@ -91,13 +91,6 @@
       <version>2.1</version>
     </dependency>
     <dependency>
-      <groupId>com.sun</groupId>
-      <artifactId>tools</artifactId>
-      <version>1.4.2</version>
-      <scope>system</scope>
-      <systemPath>${toolsjar}</systemPath>
-    </dependency>
-    <dependency>
       <groupId>javax.servlet</groupId>
       <artifactId>servlet-api</artifactId>
       <version>2.3</version>
@@ -105,28 +98,4 @@
     </dependency>
   </dependencies>
 
-  <profiles>
-    <profile>
-      <id>default-profile</id>
-      <activation>
-        <file>
-          <exists>${java.home}/../lib/tools.jar</exists>
-        </file>
-      </activation>
-      <properties>
-        <toolsjar>${java.home}/../lib/tools.jar</toolsjar>
-      </properties>
-    </profile>
-    <profile>
-      <id>mac-profile</id>
-      <activation>
-        <file>
-          <exists>${java.home}/../Classes/classes.jar</exists>
-        </file>
-      </activation>
-      <properties>
-        <toolsjar>${java.home}/../Classes/classes.jar</toolsjar>
-      </properties>
-    </profile>
-  </profiles>
 </project>

--- a/webstart-maven-plugin/src/it/it006/pom.xml
+++ b/webstart-maven-plugin/src/it/it006/pom.xml
@@ -16,4 +16,16 @@
     <module>webstart</module>
   </modules>
 
+  <build>
+    <pluginManagement>
+      <plugins>
+		<plugin>
+			<groupId>org.apache.maven.plugins</groupId>
+			<artifactId>maven-compiler-plugin</artifactId>
+			<version>3.8.1</version>
+		</plugin>
+      </plugins>
+    </pluginManagement>
+  </build>
+
 </project>

--- a/webstart-maven-plugin/src/it/it006/sign-one/pom.xml
+++ b/webstart-maven-plugin/src/it/it006/sign-one/pom.xml
@@ -19,11 +19,11 @@
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>keytool-maven-plugin</artifactId>
-        <version>1.0</version>
+        <version>1.5</version>
         <executions>
            <execution>
               <goals>
-                 <goal>genkey</goal>
+                 <goal>generateKeyPair</goal>
               </goals>
               <phase>validate</phase> <!-- should really be pre-package... -->
            </execution>
@@ -37,7 +37,8 @@
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-jar-plugin</artifactId>
+        <artifactId>maven-jarsigner-plugin</artifactId>
+		<version>3.0.0</version>
         <executions>
           <execution>
             <goals>

--- a/webstart-maven-plugin/src/it/it009/pom.xml
+++ b/webstart-maven-plugin/src/it/it009/pom.xml
@@ -96,13 +96,6 @@
       <version>1.0.5</version>
     </dependency>
     <dependency>
-      <groupId>com.sun</groupId>
-      <artifactId>tools</artifactId>
-      <version>1.4.2</version>
-      <scope>system</scope>
-      <systemPath>${toolsjar}</systemPath>
-    </dependency>
-    <dependency>
       <groupId>javax.servlet</groupId>
       <artifactId>servlet-api</artifactId>
       <version>2.3</version>
@@ -110,28 +103,4 @@
     </dependency>
   </dependencies>
 
-  <profiles>
-    <profile>
-      <id>default-profile</id>
-      <activation>
-        <file>
-          <exists>${java.home}/../lib/tools.jar</exists>
-        </file>
-      </activation>
-      <properties>
-        <toolsjar>${java.home}/../lib/tools.jar</toolsjar>
-      </properties>
-    </profile>
-    <profile>
-      <id>mac-profile</id>
-      <activation>
-        <file>
-          <exists>${java.home}/../Classes/classes.jar</exists>
-        </file>
-      </activation>
-      <properties>
-        <toolsjar>${java.home}/../Classes/classes.jar</toolsjar>
-      </properties>
-    </profile>
-  </profiles>
 </project>

--- a/webstart-maven-plugin/src/it/settings.xml
+++ b/webstart-maven-plugin/src/it/settings.xml
@@ -15,7 +15,6 @@
 -->
 
 <settings>
-
   <profiles>
     <profile>
       <id>it-repo</id>

--- a/webstart-maven-plugin/src/test/java/org/codehaus/mojo/webstart/AbstractJnlpMojoTest.java
+++ b/webstart-maven-plugin/src/test/java/org/codehaus/mojo/webstart/AbstractJnlpMojoTest.java
@@ -1,5 +1,9 @@
 package org.codehaus.mojo.webstart;
 
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
 /*
  * Licensed to the Apache Software Foundation (ASF) under one
  * or more contributor license agreements.  See the NOTICE file
@@ -28,12 +32,10 @@ import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.testing.AbstractMojoTestCase;
 import org.apache.maven.profiles.DefaultProfileManager;
 import org.apache.maven.profiles.ProfileManager;
+import org.apache.maven.project.DefaultProjectBuilder;
+import org.apache.maven.project.DefaultProjectBuildingRequest;
 import org.apache.maven.project.MavenProject;
-import org.apache.maven.project.MavenProjectBuilder;
-
-import java.io.File;
-import java.util.ArrayList;
-import java.util.List;
+import org.apache.maven.project.ProjectBuilder;
 
 /**
  * @author <a href="jerome@coffeebreaks.org">Jerome Lacoste</a>
@@ -107,7 +109,7 @@ public abstract class AbstractJnlpMojoTest
     private void setUpProject( File pomFile, AbstractMojo mojo )
         throws Exception
     {
-        MavenProjectBuilder projectBuilder = (MavenProjectBuilder) lookup( MavenProjectBuilder.ROLE );
+        DefaultProjectBuilder projectBuilder = (DefaultProjectBuilder) lookup( DefaultProjectBuilder.class );
 
         ArtifactRepositoryFactory artifactRepositoryFactory =
             (ArtifactRepositoryFactory) lookup( ArtifactRepositoryFactory.ROLE );
@@ -120,9 +122,14 @@ public abstract class AbstractJnlpMojoTest
             artifactRepositoryFactory.createArtifactRepository( "local", localRepoUrl, new DefaultRepositoryLayout(),
                                                                 policy, policy );
 
-        ProfileManager profileManager = new DefaultProfileManager( getContainer() );
-
-        MavenProject project = projectBuilder.buildWithDependencies( pomFile, localRepository, profileManager );
+//        ProfileManager profileManager = new DefaultProfileManager( getContainer() );
+        
+        final DefaultProjectBuildingRequest configuration = new DefaultProjectBuildingRequest();
+        configuration.setLocalRepository(localRepository);
+        
+        MavenProject project = projectBuilder.build(pomFile, configuration).getProject();
+        
+//        MavenProject project = projectBuilder.buildWithDependencies( pomFile, localRepository, profileManager );
 
         //this gets the classes for these tests of this mojo (exec plugin) onto the project classpath for the test
         project.getBuild().setOutputDirectory( new File( "target/test-classes" ).getAbsolutePath() );

